src/monitoring_service/service.py

```diff
@@ -163,16 +163,41 @@ def _process_new_blocks(self, last_block: BlockNumber) -> None:
 
         # check triggered events
         # TODO: create a priority queue for this
-        to_remove = []
+        events_to_remove = []
         for scheduled_event in self.context.scheduled_events:
             event = scheduled_event.event
 
             if last_block >= scheduled_event.trigger_block_number:
-                to_remove.append(scheduled_event)
+                events_to_remove.append(scheduled_event)
                 handle_event(event, self.context)
 
-        for d in to_remove:
-            self.context.scheduled_events.remove(d)
+        for event in events_to_remove:
+            self.context.scheduled_events.remove(event)
 
         if self.context.scheduled_events:
-            log.info('Scheduled_events', events=self.context.scheduled_events)
+            log.debug('Scheduled_events', events=self.context.scheduled_events)
+
+        # check pending transactions
+        # this is done here so we don't have to block waiting for receipts in the state machine
+        hash_to_remove = []
+        for tx_hash in self.context.waiting_transactions:
+            receipt = self.web3.eth.getTransactionReceipt(tx_hash)
```

Comment:
Will this return soon in all cases, or might there be cases where we wait for the receipt for many blocks, because the transaction is not mined quickly?

Reply:
This might take a while. Do you have some improvement in mind?

Reply:
I mixed it up with `waitForTransactionReceipt`. This should be ok as it is now.